packages/uploadthing/src/sdk/utils.ts

```diff
@@ -33,215 +40,166 @@ export function guardServerOnly() {
-      contentDisposition: data.contentDisposition,
-      acl: data.acl,
-    }),
-  });
-
-  if (!res.ok) {
-    const error = await UploadThingError.fromResponse(res);
-    logger.debug("Failed getting presigned URLs:", error);
-    throw error;
-  }
-
-  const json = await res.json<{
-    data: PresignedURLs;
-  }>();
-
-  logger.debug("Got presigned URLs:", json.data);
-  logger.debug("Starting uploads...");
-
-  // Upload each file to S3 in chunks using multi-part uploads
-  const uploads = await Promise.allSettled(
-    data.files.map(async (file, i) => {
-      const presigned = json.data[i];
-
-      if (!presigned) {
-        logger.error(
-          "Failed to generate presigned URL for file:",
-          file,
-          presigned,
-        );
-        throw new UploadThingError({
-          code: "URL_GENERATION_FAILED",
-          message: "Failed to generate presigned URL",
-          cause: JSON.stringify(presigned),
-        });
-      }
-
-      if ("urls" in presigned) {
-        await uploadMultipart(file, presigned, { ...opts });
-      } else {
-        await uploadPresignedPost(file, presigned, { ...opts });
-      }
-
-      // Poll for file to be available
-      logger.debug("Polling for file data...");
-      await pollForFileData({
-        url: generateUploadThingURL(`/api/pollUpload/${presigned.key}`),
-        apiKey: opts.utRequestHeaders["x-uploadthing-api-key"],
-        sdkVersion: UPLOADTHING_VERSION,
-        fetch: opts.fetch,
-      });
-      logger.debug("Polling complete.");
-
-      return {
-        key: presigned.key,
-        url: presigned.fileUrl,
-        name: file.name,
-        size: file.size,
-        type: file.type,
-        customId: "customId" in file ? file.customId ?? null : null,
-      };
-    }),
-  );
-
-  logger.debug("All uploads complete, aggregating results...");
-
-  return uploads.map((upload) => {
-    if (upload.status === "fulfilled") {
-      const data: UploadedFileData = upload.value;
-      return { data, error: null };
-    }
-    // We only throw UploadThingErrors, so this is safe
-    const reason = upload.reason as UploadThingError;
-    const error: SerializedUploadThingError = UploadThingError.toObject(reason);
-    return { data: null, error };
-  });
+type UploadFilesInternalOptions = {
+  files: FileEsque[];
+  metadata: Json;
+  contentDisposition: ContentDisposition;
+  acl: ACL | undefined;
 };
 
-async function uploadMultipart(
-  file: FileEsque,
-  presigned: MPUResponse,
-  opts: {
-    fetch: FetchEsque;
-    utRequestHeaders: Record<string, string>;
-  },
-) {
-  logger.debug(
-    "Uploading file",
-    file.name,
-    "with",
-    presigned.urls.length,
-    "chunks of size",
-    presigned.chunkSize,
-    "bytes each",
+export const uploadFilesInternal = (input: UploadFilesInternalOptions) =>
+  getPresignedUrls(input).pipe(
+    Effect.andThen((presigneds) =>
+      // TODO: Catch errors for each file and return data like
```

Comment:
Outdated comment? Seems like it is doing this?
![image](https://github.com/pingdotgg/uploadthing/assets/11494384/39397fdb-d8ef-4247-b55d-3f2a5ae721d1)

Or do you mean it should be:

`Effect.Effect<{ data, error: null }, { data: null, error }, FetchContextTag>`